core/domain/query_jobs_one_off.py

```diff
@@ -0,0 +1,143 @@
+# coding: utf-8
+#
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs to execute and get result of a query."""
+import ast
+import datetime
+
+from core import jobs
+from core.domain import exp_services
+from core.domain import query_domain
+from core.platform import models
+
+(user_models, job_models, exp_models) = (
+    models.Registry.import_models(
+        [models.NAMES.user, models.NAMES.job, models.NAMES.exploration]))
+
+
+class QueryOneOffJob(jobs.BaseMapReduceJobManager):
+    """One-off job for excuting query with given query parameters.
+    For each user we check if he/she satisfies query criteria. If user satisfies
+    query criteria then yeilds a tuple of (query_id, user_id). Reducer function
```

Comment:
If the user satisfies the query criteria, then yield a tuple (query_id, user_id). The reducer function stores all user_ids that satisfy the query in the corresponding UserQueryModel.